lib/Bindings/Python/circt/dialects/_comb_ops_ext.py

```diff
@@ -49,17 +60,45 @@ def BinaryOp(base):
   class _Class(base):
 
     @classmethod
-    def create(cls, result_type=None, **kwargs):
-      if not result_type:
-        types = list(map(lambda arg: arg.type, kwargs.values()))
-        if not types:
+    def create(cls, result_type=None, lhs=None, rhs=None):
+      if not isinstance(result_type, Type):
+        if not lhs and not rhs:
           raise TypeError("result type must be specified")
-        all_equal = reduce(lambda t1, t2: t1 == t2, types)
-        if not all_equal:
-          raise TypeError(
-              f"expected same input port types, but received {types}")
-        result_type = types[0]
-      return BinaryOpBuilder(cls, result_type, kwargs)
+        rhs = lhs
+        lhs = result_type
```

Comment:
`result_type` is inaccurately named if used this way. Maybe put it at the end and require it to be a keyword argument?

Reply:
Yeah, I think that is a much better way.